### PR TITLE
Various small bugfixes

### PR DESCRIPTION
**Pull request by rjruigrok**
* No more bartercast debug logging
* NoCrypto fix missing key

### CODE REVIEW

File: Tribler/Core/CacheDB/Notifier.py [hunk 1]
```diff
@@ -69,6 +68,7 @@ def remove_observers(self):
             self.observers = []
 
     def notify(self, subject, changeType, obj_id, *args):
+        from Tribler.Core.Utilities.twisted_thread import callInThreadPool
```
Comment (whirm) [inline Tribler/Core/CacheDB/Notifier.py]: hm, I didn't think of that... wait a sec, I'll move the utility functions out of the twisted_thread module instead.

Comment (whirm) [inline Tribler/Core/CacheDB/Notifier.py]: Can you rebase once this one is merged? https://github.com/Tribler/tribler/pull/1492

Comment (rjruigrok) [inline Tribler/Core/CacheDB/Notifier.py]: :+1:

Comment (LipuFei) [inline Tribler/Core/CacheDB/Notifier.py]: @rjruigrok #1492 has been merged.

**Comment by rjruigrok** [comment]
@whirm Thanks! I rebased. 

PR is ready if it passes.

**Comment by rjruigrok** [comment]
retest this please

**Comment by whirm** [comment]
@rjruigrok it timed out. are the tests taking more than before?

**Comment by whirm** [comment]
@rjruigrok :http://jenkins.tribler.org/job/GH_Tribler_pull-request-tester-turbo_devel/282/artifact/output/test_runners_output/00009.err/*view*/

**Comment by rjruigrok** [comment]
@whirm Not that I know of, did not change anything that could cause a test to take longer. Maybe there was a higher load on the jenkins server?

**Comment by whirm** [comment]
I doubt it very much (the gorilas have either 16 or 24 cores each). But even so, the test shouldn't get stuck there for 15 mins.

**Comment by rjruigrok** [comment]
Can we see which test exactly took so long? I doubt it has to do with the changes in this PR, did this also happen in other PR's too?

**Comment by whirm** [comment]
@rjruigrok I added a link to the log already:

> @rjruigrok :http://jenkins.tribler.org/job/GH_Tribler_pull-request-tester-turbo_devel/282/artifact/output/test_runners_output/00009.err/*view*/

It timed out while running: 

```test_hidden_services_with_exit_nodes (Tribler.Test.test_hidden_community.TestHiddenCommunity) ... ```

**Comment by rjruigrok** [comment]
Yup saw those, but the timeout occured while running `test_hidden_services_with_exit_nodes` does not say that the problem is in this test. Possibly it is caused by one of the earlier test(s). 
Is it possible to add timestamps in the test logging?

**Comment by whirm** [comment]
nope, that's nose printing the info. You can look at non-timed out runs' reports to see how much each test usually takes. But the thing here is that the tests are not timing out after a reasonable amount of time by themselves.

**Comment by whirm** [comment]
@rjruigrok I don't know if it's you who wrote the tests or if you are responsible of fixing them. So tell me if you aren't :)

**Comment by rjruigrok** [comment]
The tests should be timing out themselves and it looks like it's not happening. I'll have a look at this. :)

**Comment by whirm** [comment]
@rjruigrok are you doing it in this branch or it's ready to merge?

**Comment by rjruigrok** [comment]
@whirm I reduced the CallConditional timeouts significantly to around 15s above the time a test usually takes. Maybe to rigid, but this can be adjusted in small amounts if it's necessary. Let's see how it performs after a number of retests

**Comment by rjruigrok** [comment]
retest this please

**Comment by NielsZeilemaker** [comment]
@rjruigrok the CallConditional timeouts don't really impact the runtime. Those will continue if the condition is true, I would attempt to convert the Call methods into a CallConditional. That would speedup the tests.

**Comment by rjruigrok** [comment]
@NielsZeilemaker Hmm good point. There are a number of Calls and sleeps in the code, those can be eliminated completely. 

But as you can see in test 282 it timed out after 15 minutes of execution without any feedback, that should not happen, by having a lower CallConditional's this hopefully will not happen anymore :)

**Comment by rjruigrok** [comment]
retest this please

**Comment by rjruigrok** [comment]
@NielsZeilemaker I thought there were some unnecessary Call methods around, but they actually are there most of the times to get a scenario-like running of the test. for example the negative tests start a download, waits for 30 seconds using a Call(30, ...) call to ensure there is no progress after 30 seconds.

**Comment by rjruigrok** [comment]
Is there still need to have the `test_anon_tunnel` test here? It's just tunneling "42" through a mocked socks. The correct working of circuits is already validated by the `test_anon_download` test. It saves another 57 seconds of testing.. :smiley:

**Comment by NielsZeilemaker** [comment]
I would keep it, I guess its still a valid test.

**Comment by rjruigrok** [comment]
True. 
Then this PR is ready in my opinion. Tests are running stable again.

**Pull request merged**